api/src/test/java/org/apache/iceberg/TestSchema.java

```diff
@@ -95,14 +140,21 @@ public void testUnsupportedInitialDefault(int formatVersion) {
             formatVersion);
   }
 
-  @Test
-  public void testSupportedInitialDefault() {
-    assertThatCode(() -> Schema.checkCompatibility(INITIAL_DEFAULT_SCHEMA, 3))
+  private static int[] testSupportedInitialDefault =
+      IntStream.rangeClosed(DEFAULT_VALUES_MIN_FORMAT_VERSION, MAX_FORMAT_VERSION).toArray();
+
+  @ParameterizedTest
+  @FieldSource
+  public void testSupportedInitialDefault(int formatVersion) {
+    assertThatCode(() -> Schema.checkCompatibility(INITIAL_DEFAULT_SCHEMA, formatVersion))
         .doesNotThrowAnyException();
   }
 
+  private static int[] testSupportedWriteDefault =
```

Comment:
```suggestion
  private static int[] supportedWriteDefault =
```

I would also move this to the top